analysis/quantifier-comparisons/06-alevin-fry-vs-cellranger.Rmd

```diff
@@ -0,0 +1,769 @@
+  ylab("UMI/cell") + 
+  xlab("") + 
+  ylim(c(0,50000))
+```
+
+```{r fig.width = 5, fig.height=3}
+ggplot(nucleus_qc_common, aes(x = alevin_resolution, y = detected, fill = alevin_alignment)) + 
+  geom_boxplot() + 
+  facet_grid(~ sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Genes detected/cell") + 
+  xlab("") +
+  coord_cartesian(ylim = c(0,15000))
+```
+
+```{r fig.height=3, fig.width=5}
+ggplot(cell_qc_common, aes(x = alevin_resolution, y = detected, fill = alevin_alignment)) + 
+  geom_boxplot() + 
+  facet_grid(sample ~ index_type) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Genes detected/cell") + 
+  xlab("") +
+  coord_cartesian(ylim = c(0,10000))
+```
+Overall, it looks like we have eliminated many of the differences between the tools in the single-nuclei RNA-seq data. 
+Although it appears that across the board the cr-like resolution gives more on par distributions to cellranger than the full resolution. 
+The `full` resolution seems to have higher UMIs/cell and genes/cell across most samples. 
+Another consistent finding is that the `--sketch` or pseudoalignment is giving higher UMIs/cell and genes detected/cell across the board, although this is more noticeable in the single-nuclei samples. 
+It also seems to be more apparent with the `full` resolution which is probably to be expected since the `full` resolution is not throwing out multi-mapped genes like the `cr-like` resolution will.  
+
+It does look like alevin-fry is detecting more cells than cellranger, so perhaps there is something we could be doing at the filtering stage that could help improve this? 
+
+Let's take a quick look at the number of cells detected by tool just to be sure that this hypothesis is correct. 
+```{r fig.height= 2, fig.width=5}
+cell_numbers <- coldata_info_df %>% 
+  dplyr::group_by(quant_id, tool, sample) %>% 
+  #dplyr::filter(sample != "SCPCR000118") %>%
+  dplyr::tally()
+
+ggplot(cell_numbers, aes(x = sample, y = n, color = tool)) + 
+  geom_point() +
+  theme_classic()
+```
+As expected, for all samples (except SCPCR000006), cellranger is detecting fewer cells than in alevin-fry. 
+I'm also not sure what has happened with SCPCR000118, but we saw this previously and it also shows strange patterns in mitochondrial content and very low coverage across the board.
+
+
+## Per Cell QC Metrics in Shared Cells, Minimum Gene Coverage 
+
+To see if the increase in genes detected/cell in the full resolution could be due to low covered genes, let's look at the number of genes detected if we were to remove genes only found in < 5% of cells. 
+
+
+```{r}
+# use addCellQC function to addPerCellQC with gene detection threshold
+# only genes with detection in > 5% of cells will be included in the QC calculations
+sce_list <- sce_list %>%
+  purrr::map(.f =  
+               ~ purrr::walk(.x, addCellQC, mito = mito_genes, threshold = 5
+  ))
```

Comment:
You will need `purrr::map()` in the internal function or the changed results will not be returned. Since you are piping into the outer `map`,  you don't need to specify the `.f`. 
```suggestion
sce_list <- sce_list %>%
  purrr::map(
    ~ purrr::map(.x, addCellQC, mito = mito_genes, threshold = 5)
  )
```